libats-messaging/src/main/scala/com/advancedtelematic/libats/messaging/kafka/KafkaMessage.scala

```diff
@@ -0,0 +1,33 @@
+package com.advancedtelematic.libats.messaging.kafka
+
+import akka.Done
+import akka.kafka.ConsumerMessage.CommittableOffsetBatch
+import com.advancedtelematic.libats.messaging.MessageListener.{CommittableMsg, KafkaMsg}
+
+import scala.concurrent.Future
+
+trait Commiter {
+  def commit(msg: CommittableMsg[_]): Future[Done]
+
+  def commit(msgs: Seq[CommittableMsg[_]]): Future[Done]
+}
+
+class KafkaCommiter extends Commiter {
+  override def commit(msg: CommittableMsg[_]): Future[Done] = msg match {
+    case kafkaMsg: KafkaMsg[_] =>
+      kafkaMsg.offset.commitScaladsl()
+    case _ =>
+      Future.failed(new RuntimeException("fatal error: Kafka committer can only commit KafkaMsg"))
+  }
+
+  override def commit(msgs: Seq[CommittableMsg[_]]): Future[Done] = {
+    val offset = msgs.foldRight(CommittableOffsetBatch.empty) {
+      case (kafkaMsg: KafkaMsg[_], acc) =>
```

Comment:
I tried many different things and I cannot come up with a better solution without adding an unreasonable amount of polymorphism. Even then I am not 100% it would work.